beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java

```diff
@@ -63,20 +63,10 @@ public ExpiredMetadataHandler(HousekeepingMetadataRepository housekeepingMetadat
    */
   @Override
   public void cleanupMetadata(HousekeepingMetadata housekeepingMetadata, LocalDateTime instant, boolean dryRunEnabled) {
-    if (dryRunEnabled) {
-      cleanup(housekeepingMetadata, instant, dryRunEnabled);
-    } else {
-      cleanupAndUpdate(housekeepingMetadata, instant, dryRunEnabled);
-    }
-  }
-
-  private void cleanupAndUpdate(HousekeepingMetadata housekeepingMetadata, LocalDateTime instant,
-      boolean dryRunEnabled) {
+    metadataCleaner.init();
```

Comment:
inject the HiveClientFactory hiveClientFactory in the constructor then here you can do:

try (HiveClient client =  hiveClientFactory.newInstance()) {
//body of this whole method

}